### DISCUSSION TITLE
Use github release instead of tag pushing

### DISCUSSION THREAD

**Discussion by ssbarnea** [tox-dev/tox#2339] [Ideas]
Please use only github release page to create new released (and tags) instead of pushing tags directly. This allow you to include some release notes and more importantly it gives you the option to start a discussion thread for each release, where you can receive feedback from that particular release.

For example `4.0.0b1` was created using git push, so is not visible on https://github.com/tox-dev/tox/releases and it is much harder to provide feedback on it. Keep in mind that having one thread per release is a very good way to gather feedback on how good or bad it was.  Anyone visiting github will only see `4.0.0a10` as last release, today's beta being hidden from view due to the way it was released.

A hint: you can still create that release even if the tag was already created before it.

Digging the issue tracker in order to find out bugs related to particular release is difficult and imprecise. Also that approach does not make it possible to identify positive feedback on that release. Discussions threads sorts both.

**Comment by gaborbernat** [comment]
Sadly this is against our release pipeline. So I'm -1 on this as I'd rather not add GitHub as release dependency. Changes are part of the changelog section in our documentation and I'd rather not duplicate it in GitHub release page. We don't use GitHub releases in general.

**Reply by ssbarnea** [reply]
Something does not adds-up here because I do see the other releases being created, only that one was just a tag.

I do not care much if you first create the tag and the release after, even manually. Even the release note content is less important than **marking last release visible in github project main page** and providing a **discussion thread** for it.

I doubt you want to hide releases from github, as that will cause lots of reports from confused visitors.

**Reply by gaborbernat** [reply]
We usually did it manually afterwards, but as you can see it's easy to miss it. I really wish Github would consider tags releases.

**Reply by ssbarnea** [reply]
Two problems with current beta which are a deal breaker for pushing it:
* no `tox` clie
* no `tox -va` alias, while `tox list` I cannot retrain my muscle memory, we really need a compatibility mode for some time, especially as there is no way to add a shell alias to make it work.

Keep in mind that it will take a long time (likely even more than a year) till tox 4 will become mainstream. Running `tox -a` nad `tox -va` is very common, almost as common as `tox -e foo`. Other options are not important to be kept compatible, but if we do not keep those, there will be pain, lot of it.

**Reply by gaborbernat** [reply]
`tox4 -a` or `tox4 -va` should work just fine. If it doesn't it's a bug 😢

**Reply by gaborbernat** [reply]
We've added some changes to address your first concern. The second is an odd one because `tox -av` works but `tox -va` does not, that's TBD.
